### PR TITLE
Use GroupVersionResource in ApplicationSelector

### PR DESCRIPTION
**Pull request by baijum**
Fixes https://issues.redhat.com/browse/APPSVC-324

### Motivation

Be consistent with Kubernetes APIs

### Changes

Use GroupVersionResource from Kubernetes.

### Testing

WIP

### CODE REVIEW

File: deploy/crds/apps.openshift.io_servicebindingrequests_crd.yaml [hunk 1]
```diff
@@ -94,6 +94,7 @@ spec:
                 version:
```
Comment (DhritiShikhar) [inline deploy/crds/apps.openshift.io_servicebindingrequests_crd.yaml]: Shouldn't `version` and `resource` be removed from `properties`?

**Comment by DhritiShikhar** [comment]
@baijum Can you update the README and CR as well according to this change?

**Comment by DhritiShikhar** [comment]
PR looks okay. 

e2e tests are failing which need to be updated:

```
--- FAIL: TestAddSchemesToFramework (131.88s)
    servicebindingrequest_test.go:69: Adding ServiceBindingRequestList scheme to cluster...
    servicebindingrequest_test.go:73: Adding ClusterServiceVersionList scheme to cluster...
    servicebindingrequest_test.go:77: Adding DatabaseList scheme to cluster...
    servicebindingrequest_test.go:81: Adding EtcdCluster scheme to cluster...
    --- FAIL: TestAddSchemesToFramework/end-to-end (127.09s)
        --- FAIL: TestAddSchemesToFramework/end-to-end/scenario-etcd-unannotated-app-db-sbr (20.41s)
            servicebindingrequest_test.go:147: Creating a new test context...
            servicebindingrequest_test.go:128: Initializing cluster resources...
            servicebindingrequest_test.go:138: Using namespace 'test-namespace-d38f427e-4cd3-4c15-8bf6-90a3b015ffdf' for testing...
            servicebindingrequest_test.go:504: Starting end-to-end tests for operator, using suffix '6129484611666145821'!
            servicebindingrequest_test.go:324: Creating Deployment mock object 'types.NamespacedName{Namespace:"test-namespace-d38f427e-4cd3-4c15-8bf6-90a3b015ffdf", Name:"e2e-application-6129484611666145821"}'...
skipped 17 lines unfold_more
                	            	ServiceBindingRequest.apps.openshift.io "e2e-service-binding-request-6129484611666145821" is invalid: [spec.applicationSelector.group: Required value, spec.applicationSelector.resource: Required value, spec.applicationSelector.version: Required value]
```

**Comment by baijum** [comment]
Not sure why lint fails.

Found something related: https://github.com/pypa/pip/issues/7217

File: deploy/crds/apps.openshift.io_servicebindingrequests_crd.yaml [hunk 2]
```diff
@@ -94,10 +94,7 @@ spec:
...
                   type: string
               required:
-              - group
```
Comment (DhritiShikhar) [inline deploy/crds/apps.openshift.io_servicebindingrequests_crd.yaml]: @baijum Why did you remove `gvr` from required?

Comment (baijum) [reply deploy/crds/apps.openshift.io_servicebindingrequests_crd.yaml]: Reverted. 83750c1b4731363693d85af45ad30ffb89027f78

I am not sure why the test is failing this was just an experiment.

Comment (DhritiShikhar) [reply deploy/crds/apps.openshift.io_servicebindingrequests_crd.yaml]: ok
